supervision/utils/internal.py

```diff
@@ -141,3 +142,42 @@ def __get__(self, owner_self: object, owner_cls: type) -> object:
         The result of calling the function stored in 'fget' with 'owner_cls'.
         """
         return self.fget(owner_cls)
+
+
+def get_instance_variables(cls: Any, include_properties=False) -> Set[str]:
```

Comment:
Check if you get class or instance of class. And raise with some descriptive message if it is a class.